### ISSUE TITLE
Support string-comparison queries

### ISSUE THREAD

**Issue by nolanamy** [realm/realm-java#4385]
I need to do `between`/`lessThan`/`greaterThan` queries on `String` fields.

I'd like to use it to grab a series of alphabetically-sorted chunks from a collection.

If we can sort on a field (as we can with `String` fields), then we should be able to comparison-query on it too... right?

**Comment by kneth** [comment]
@nolanamy I can see the value in your suggestion, and that you don't wish to sort all of objects to iterate through then to find only a fraction. Realm Core doesn't expose an API for doing it directly.

**Comment by Grohden** [comment]
no news on this? :/

**Comment by kneth** [comment]
@Grohden Not really. We have looked at supporting the same string-based query language as Realm JS is using (see #6116) but Realm Core still lacks this functionality.

**Comment by caeduk** [comment]
@bmunkholm @kneth still no news? (5 years anniversary on this issue)

**Comment by rorbech** [comment]
@caeduk There is currently no typesafe API (`lessThan`, `greaterThan`, etc.) around this, but you can use [`rawPredicate`](https://www.mongodb.com/docs/realm/sdk/java/api/io/realm/RealmQuery/#rawpredicate) to compare these. But please note that there are some [caveats with the current implementation](https://github.com/realm/realm-core/issues/6120).

**Comment by cmelchior** [comment]
@caeduk This has been added. Unfortunately, the paper trail is a bit cold as to when exactly when, but it is a least present in 10.12.0. Closing this now.
